normandy/selfrepair/static/js/self_repair.js

```diff
@@ -0,0 +1,22 @@
+import Normandy from './normandy_driver.js';
+import {fetchRecipes, filterContext, doesRecipeMatch, runRecipe} from './self_repair_runner.js';
+
+// Actually fetch and run the recipes.
+fetchRecipes().then(recipes => {
+  let context = filterContext().then(context => {
```

Comment:
I changed filterContext() to return a promise, rather than just the context object. The way it was [set up earlier](https://github.com/Osmose/normandy/blob/776acddfff1a33f1c65b213b22da1f6c0195dd41/normandy/selfrepair/static/js/self_repair_runner.js#L121-L127), the context object would be returned with the `normandy` property === a promise that was still pending.